src/conf_mode/protocols_bgp.py

```diff
@@ -103,6 +103,10 @@ def verify(bgp):
                 if 'ebgp_multihop' in peer_config and 'ttl_security' in peer_config:
                     raise ConfigError('You can\'t set both ebgp-multihop and ttl-security hops')
 
+                # Check spaces in the password
+                if ' ' in peer_config['password']:
```

Comment:
Assuming a user has no `password` specified thus there will be no `password` key in the `peer_config` dict which will then throw a KeyError.

A better solution would be: `if 'password' in peer_config and ' ' in peer_config['password']:`

Reply:
Thanks, fixed.